xarray/tests/test_dataset.py

```diff
@@ -3431,6 +3431,47 @@ def test_expand_dims_kwargs_python36plus(self) -> None:
         )
         assert_identical(other_way_expected, other_way)
 
+    @pytest.mark.parametrize("create_index_flag", [True, False])
+    def test_expand_dims_create_index_data_variable(self, create_index_flag):
+        # data variables should not gain an index ever
+        ds = Dataset({"x": 0})
+        expanded = ds.expand_dims("x", create_index=create_index_flag)
```

Comment:
Shouldn't this raise a warning for `create_index=True` since that option is effectively ignored.

Reply:
Why would that raise a warning? You think a kwarg that is irrelevant when passed should raise a warning? 

I also feel like that could create unnecessary warnings, especially if we made a global `create_index` flag later like was discussed in https://github.com/pydata/xarray/pull/8872#issuecomment-2030406780

Reply:
Yes, I think that if a user explicitly writes `ds.expand_dims("x", create_index=True)` and we don't create an index, we should either warn or error.

In this case, it seems OK to warn and ask the user to call `set_coords("x")` first. We want to encourage that behaviour anyway.

I have not thought about the global option yet, but usually we handle that by setting `None` as the default value and then harmonizing with the global setting.

Reply:
I'm also OK punting that to a future PR. The current behaviour is pretty broken already.

Reply:
Added a `UserWarning` in https://github.com/pydata/xarray/pull/8960/commits/7e8f895c2746e2a75979b5675967b434aef09ee6.